.github/workflows/on-pullrequest-dagster.yml

```diff
@@ -0,0 +1,66 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+name: On Pull Request - Dagster
+
+on:
+  pull_request:
+    branches: [ main ]
+    paths:
+      - "src/dagster_data_lake/**/*.py"
+      - "**/pyproject.toml"
+      - ".github/**/*.yml"
+  workflow_dispatch:
+
+jobs:
+  run-py-tests:
+    name: Run Python Tests
+    runs-on: ubuntu-latest
+
+    defaults:
+      run:
+        working-directory: src
+
+    env:
+      PROJECT: dagster_data_lake
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # v3.0.2
+
+    - name: Install Poetry
+      run: pipx install poetry
+
+    - name: Install Python 3.9
+      uses: actions/setup-python@98f2ad02fd48d057ee3b4d4f66525b231c3e52b6 # v3.1.2
+      with:
+        python-version: "3.9"
+        cache: "poetry"
+
+    - name: Install Dependencies
+      run: |
+        # Next line keeps poetry from trying to use the runner's default Python version
+        pushd ${{ env.SRC_DIR }} && poetry env use "3.9" && popd
+        poetry install
+
+    - name: MyPy Linting
+      run: poetry run mypy ${{ env.PROJECT }} --junit-xml mypy-junit.xml
+
+    - name: Flake8 Linting
+      run: |
+        poetry run flake8 ${{ env.PROJECT }} --output-file flake8.txt
+        poetry run flake8_junit flake8.txt flake8.xml
+
+    # This needs some work. Stephen's not sure how to setup the tests correctly.
+    # - name: Run Unit Tests
+    #   run: |
+    #     poetry run pytest ${{ env.PROJECT }} --junitxml=pytest-junit.xml
```

Comment:
@jegasape I see that there are some tests for the Dagster project. Are they unit tests, running in complete isolation, or are they integration tests that require having an Ed-Fi API running?

Reply:
We currently have the infrastructure, after our call we will work on creating test to validate the data and the parquet files.